### PR TITLE
feat(ProgressIndicator): add experimental progress indicator

### PR DESCRIPTION
**Pull request by emyarod**
Closes IBM/carbon-components-react#1647

depends on https://github.com/IBM/carbon-components/pull/1612

This PR adds experimental progress indicator styles and features. In order to support the overflow tooltip functionality, the `<Tooltip>` component requires a style override since the existing CSS classes on the React `<Tooltip>` component conflict with the `<ProgressIndicator>` overflow tooltip's styles

specifically referring to these two lines:

https://github.com/IBM/carbon-components-react/pull/1726/files#diff-24f3b48f435eda51789be323bce053c5R98

and 

https://github.com/IBM/carbon-components-react/pull/1726/files#diff-7efca1118843036947e04d876136f7c6R421

### CODE REVIEW

File: src/components/Tooltip/Tooltip.js [hunk 1]
```diff
@@ -418,6 +418,7 @@ export default class Tooltip extends Component {
               onBlur={evt => this.handleMouse(evt)}
               aria-haspopup="true"
               aria-expanded={open}
+              style={this.props.style}
```
Comment (asudoh) [inline src/components/Tooltip/Tooltip.js]: Can we do (probably additional) CSS classes rather than `style` attribute?

Comment (emyarod) [reply src/components/Tooltip/Tooltip.js]: the PR is open here: https://github.com/IBM/carbon-components/pull/1635

**Comment by emyarod** [comment]
dependent on https://github.com/IBM/carbon-components/pull/1635

File: src/components/ProgressIndicator/ProgressIndicator.js [hunk 2]
```diff
@@ -9,49 +9,113 @@ import PropTypes from 'prop-types';
...
+          showIcon={false}
+          triggerClassName={`${prefix}--progress-label`}
+          triggerText={label}
+          {...overflowTooltipProps}>
+          {overflowTooltipChildren}
+        </Tooltip>
+      ) : (
+        <p
+          className={`${prefix}--progress-label`}
+          aria-describedby={componentsX ? 'label-tooltip' : null}>
```
Comment (asudoh) [inline src/components/ProgressIndicator/ProgressIndicator.js]: We make sure we generate instance-speciic ID for ID refs like `aria-describedby`. Which element should this point to BTW?

File: src/components/ProgressIndicator/ProgressIndicator.js [hunk 3]
```diff
@@ -82,6 +148,36 @@ ProgressStep.propTypes = {
...
+
+  /**
+   * Provide the content of a progress step tooltip
+   */
+  overflowTooltipChildren: PropTypes.node,
+
+  /**
+   * Provide the props that describe a progress step tooltip
+   */
+  overflowTooltipProps: PropTypes.object,
```
[the remark this answers is not in the document]
Comment (emyarod) [reply src/components/ProgressIndicator/ProgressIndicator.js]: refactored to use a `renderTooltip` prop

File: src/components/ProgressIndicator/ProgressIndicator.js [hunk 4]
```diff
@@ -9,49 +9,98 @@ import PropTypes from 'prop-types';
...
+      </svg>
+    );
+  })();
 
   return (
     <li className={classes}>
       {currentSvg || completeSvg || incompleteSvg}
-      <p className={`${prefix}--progress-label`}>{label}</p>
+      {componentsX && renderTooltip ? (
+        renderTooltip()
```
Comment (asudoh) [inline src/components/ProgressIndicator/ProgressIndicator.js]: * Given `renderTooltip()` is a new feature, looks that we don't need to role it off for `v10`?
* Can we set default render code as the default prop?
* Can we support stateful here as well...?

Imagining something like below, wondering if it makes sense to everybody:

```diff
--- a/src/components/ProgressIndicator/ProgressIndicator.js
+++ b/src/components/ProgressIndicator/ProgressIndicator.js
@@ -17,6 +17,8 @@ import { componentsX } from '../../internal/FeatureFlags';
 
 const { prefix } = settings;
 
+const defaultRenderLabel = props => <p {...props} />;
+
 export const ProgressStep = ({ ...props }) => {
   const {
     label,
@@ -27,7 +29,7 @@ export const ProgressStep = ({ ...props }) => {
     invalid,
     secondaryLabel,
     disabled,
-    renderTooltip,
+    renderLabel: ProgressStepLabel,
   } = props;
 
   const classes = classnames({
@@ -91,11 +93,7 @@ export const ProgressStep = ({ ...props }) => {
   return (
     <li className={classes}>
       {currentSvg || completeSvg || incompleteSvg}
-      {componentsX && renderTooltip ? (
-        renderTooltip()
-      ) : (
-        <p className={`${prefix}--progress-label`}>{label}</p>
-      )}
+      <ProgressStepLabel className={`${prefix}--progress-label`}>{label}</ProgressStepLabel>
       {componentsX &&
       secondaryLabel !== null &&
       secondaryLabel !== undefined ? (
@@ -146,7 +144,7 @@ ProgressStep.propTypes = {
    * An optional parameter to allow for overflow content to be rendered in a
    * tooltip.
    */
-  renderTooltip: PropTypes.function,
+  renderLabel: PropTypes.function,
 
   /**
    * Provide the props that describe a progress step tooltip
@@ -164,6 +162,10 @@ ProgressStep.propTypes = {
   tooltipId: PropTypes.string,
 };
 
+ProgressStep.defaultProps = {
+  renderLabel: defaultRenderLabel,
+};
+
 export class ProgressIndicator extends Component {
   state = {}; 
```

Comment (emyarod) [reply src/components/ProgressIndicator/ProgressIndicator.js]: makes sense, thanks for the suggested changes @asudoh

**Comment by asudoh** [comment]
Thank you for the update @emyarod! @joshblack Does @emyarod's latest code look good to you...? Thanks!

**Comment by asudoh** [comment]
Merging as all review comments have been addressed. Anybody don't hesitate to bring up further concerns - Thanks!

**Pull request merged**